test/integration_test.py

```diff
@@ -386,13 +386,21 @@ def test_catalog_listing(self):
 
     def test_snapshot_listing(self):
         """
-        Test with a small page size to be sure paging works
+        Test with minimal page sizes (1, 2) and a subset of deployment-specific
+        snapshots (obtain via a filtered request). To make sure paging works.
         """
-        page_size = 5
-        with mock.patch.object(TDRClient, 'page_size', page_size):
-            paged_snapshots = self._public_tdr_client.snapshot_names_by_id()
-        snapshots = self._public_tdr_client.snapshot_names_by_id()
-        self.assertEqual(snapshots, paged_snapshots)
+        deployment_specific_prefix_filter = {
```

Comment:
It is a substring filter, not a prefix filter.

```suggestion
        # Without a filter, the test takes so long that there's a real risk of
        # failure due to new snapshots being added mid-test
        snapshot_filters_by_deployment = {
```